File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
       int findMin(vector<int> &nums) {

        int l = 0, r = nums.size()-1 , ans = nums[0];
        while(l<=r){
            if (nums[l] < nums[r]) {
                ans = min(nums[l],ans);
                break;
            }
              int mid = (l+r)/2;
              ans = min(ans , nums[mid]);
              if(nums[mid]<nums[l]){
                     r = mid-1;
              }else{
                     l = mid+1;
              }
        }
        return ans;

}

};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        st = set()
        mn = float("inf")
        for num in nums:
            st.add(num)
            mn = min(mn, num)

        ans, ctr, prevNum = 0, 0, mn
        while len(st) > 0:
            if prevNum in st:
                ctr += 1
                st.remove(prevNum)
            else:
                prevNum = st.pop()
                ctr = 1
            ans = max(ans, ctr)
            prevNum += 1
        return ans
